packages/pangea-sdk/pangea/services/authz.py

```diff
@@ -0,0 +1,344 @@
+
+    def tuple_delete(self, tuples: List[Tuple]) -> PangeaResponse[TupleDeleteResult]:
+        """Delete tuples.
+
+        Delete tuples in the AuthZ Service.
+
+        Args:
+            tuples (List[Tuple]): List of tuples to be deleted.
+
+        Raises:
+            PangeaAPIException: If an API Error happens.
+
+        Returns:
+            Pangea Response with empty result.
+            Available response fields can be found in our
+            [API Documentation](https://pangea.cloud/docs/api/authz#tuple-delete).
+
+        Examples:
+            # TODO: Complete
+        """
+
+        input_data = TupleDeleteRequest(tuples=tuples)
+        return self.request.post("v1beta/tuple/delete", TupleDeleteResult, data=input_data.dict(exclude_none=True))
+
+    def check(
+        self,
+        resource: Resource,
+        action: str,
+        subject: Subject,
+        debug: Optional[bool] = None,
+        attributes: Optional[Dict[str, Union[int, str]]] = None,
+    ) -> PangeaResponse[CheckResult]:
+        """Perform a check request.
+
+        Check if a subject has permission to perform an action on the resource.
+
+        Args:
+            resource (Resource): The resource to check.
+            action (str): The action to check.
+            subject (Subject): The subject to check.
+            debug (Optional[bool]): Setting this value to True will provide a detailed analysis of the check.
+            attributes (Optional[Dict[str, Union[int, str]]]): Additional attributes for the check.
+
+        Raises:
+            PangeaAPIException: If an API Error happens.
+
+        Returns:
+            Pangea Response with the result of the check.
+            Available response fields can be found in our
+            [API Documentation](https://pangea.cloud/docs/api/authz#check).
+
+        Examples:
+            # TODO: Complete
+        """
+
+        input_data = CheckRequest(resource=resource, action=action, subject=subject, debug=debug, attributes=attributes)
+        return self.request.post("v1beta/check", CheckResult, data=input_data.dict(exclude_none=True))
+
+    def list_resources(self, namespace: str, action: str, subject: Subject) -> PangeaResponse[ListResourcesResult]:
+        """List resources.
+
+        Given a namespace, action, and subject, list all the resources in the namespace
+        that the subject has the action with.
+
+        Args:
+            namespace (str): The namespace to filter resources.
+            action (str): The action to filter resources.
+            subject (Subject): The subject to filter resources.
+
+        Raises:
+            PangeaAPIException: If an API Error happens.
+
+        Returns:
+            Pangea Response with a list of resource IDs.
+            Available response fields can be found in our
+            [API Documentation](https://pangea.cloud/docs/api/authz#list-resources).
+
+        Examples:
+            # TODO: Complete
```

Comment:
```
response = authz.check(
  namespace="file",
  action="update",
  subject=Subject(
    namespace="user",
    id="user_1"
  ),
)
```

Reply:
This is `list_resources()`, not `check()`, but I added an example based on yours.